connectors/sources/abs.py

```diff
@@ -0,0 +1,219 @@
+        logger.info("Validating ABS connection string...")
+        self._validate_connection_string()
+        try:
+            async with BlobServiceClient.from_connection_string(
+                conn_str=self.connection_string, retry_total=self.retry_count
+            ) as azure_base_client:
+                await azure_base_client.get_account_information()
+                logger.info("Successfully connected to the Azure Blob Storage.")
+        except Exception:
+            logger.exception("Error while connecting to the Azure Blob Storage.")
+            raise
+
+    def prepare_blob_doc(self, blob, container_metadata):
+        """Prepare key mappings to blob document
+
+        Args:
+            blob (dictionary): Blob document from Azure Blob Storage
+            container_metadata (string): Blob container meta data
+
+        Returns:
+            dictionary: Modified document with the help of adapter schema
+        """
+        document = {
+            "_id": f"{blob['container']}/{blob['name']}",
+            "_timestamp": blob["last_modified"].isoformat(),
+            "created at": blob["creation_time"].isoformat(),
+            "content type": blob["content_settings"]["content_type"],
+            "container metadata": str(container_metadata),
+            "metadata": str(blob["metadata"]),
+            "leasedata": str(blob["lease"]),
+        }
+        for elasticsearch_field, azure_blob_storage_field in BLOB_SCHEMA.items():
+            document[elasticsearch_field] = blob[azure_blob_storage_field]
+        return document
+
+    async def get_content(self, blob, timestamp=None, doit=None):
+        """Get blob content via specific blob client
+
+        Args:
+            blob (dictionary): Modified blob document
+            timestamp (timestamp, optional): Timestamp of blob last modified. Defaults to None.
+            doit (boolean, optional): Boolean value for whether to get content or not. Defaults to None.
+
+        Returns:
+            dictionary: Content document with id, timestamp & text
+        """
+        blob_size = int(blob["size"])
+        if not (self.enable_content_extraction and doit and blob_size):
+            return
+
+        blob_name = blob["title"]
+        if os.path.splitext(blob_name)[-1] not in TIKA_SUPPORTED_FILETYPES:
+            logger.warning(f"{blob_name} can't be extracted")
+            return
+
+        if blob_size > DEFAULT_FILE_SIZE_LIMIT:
+            logger.warning(
+                f"File size {blob_size} of file {blob_name} is larger than {DEFAULT_FILE_SIZE_LIMIT} bytes. Discarding the file content"
+            )
+            return
+
+        async with BlobClient.from_connection_string(
+            conn_str=self.connection_string,
+            container_name=blob["container"],
+            blob_name=blob_name,
+            retry_total=self.retry_count,
+        ) as blob_client:
+            data = await blob_client.download_blob()
```

Comment:
we can probably do the same work we have done in GCS to add concurrency when downloading the blob data

Reply:
As discussed for GCS, Do we still need to keep concurrency for `list_blob()` as it is not taking that much time to execute?
Related comment : https://github.com/elastic/connectors-python/issues/256#issuecomment-1373338415

Reply:
Sounds good to not add it